Kata 2.0: add metrics design documents (#665)
kata-containers/documentation: how-to/how-to-set-prometheus-and-grafana.md

* [Install Grafana](#install-grafana)
* [Setup Grafana](#setup-grafana)
  * [Create datasource](#create-datasource)
  * [Import dashboard](#import-dashboard)

Both Prometheus and Grafana can installed using pre-compiled distributions.

## Install Prometheus

Download and extract Prometheus.

```
wget https://github.com/prometheus/prometheus/releases/download/v2.18.1/prometheus-2.18.1.linux-amd64.tar.gz
tar xvfz prometheus-*.tar.gz
cd prometheus-2.18.1.linux-amd64
```

To let Prometheus to collect metrics from Kata containers, first add `magent` to a Prometheus target.

Open `prometheus.yml` and add `localhost:8090` to `scrape_configs` -> `static_configs`:

```
scrape_configs:
  - job_name: 'prometheus'
    static_configs:
      - targets: ['localhost:8090']
```

And then start Prometheus server:

```
./prometheus --config.file=prometheus.yml
```

Prometheus listen on "\*:9090" in default.

## Install Grafana

In another terminal session, download and extract Grafana.

```
wget https://dl.grafana.com/oss/release/grafana-7.0.1.linux-amd64.tar.gz
tar -zxvf grafana-7.0.1.linux-amd64.tar.gz
cd grafana-7.0.1
bin/grafana-server
```

## Setup Grafana

> After Grafana server started, you can open [`http://localhost:3000`](http://localhost:3000) to access Grafana, the default user/password is `admin/admin`.

Author: For this how-to, I think we can leave it as default. And I added Grafana settings about security by the official document.

To use Grafana show data from Prometheus, you must create a Prometheus datasoure and dashboard.

### Create datasource

Open [`http://localhost:3000/datasources/new`](http://localhost:3000/datasources/new) in your browser, select Prometheus from time series databases list.

Normally you only need to set `URL` to `http://localhost:9090` to let it work, and leave the name as `Prometheus` as default.

### Import dashboard

Here is a [sample dashboard](data/dashboard.json) provided, you can import and evaluate with it.

You can import this dashboard using [browser](http://localhost:3000/dashboard/import), or using `curl` command in console.

```
curl -XPOST -i localhost:3000/api/dashboards/import \

>     -u admin:admin \

Author: As above, may be leave it as default like easy to try this how-to.

# How to setup Prometheus and Grafana as observability platform

> Prometheus and Grafana are open sources analytics & monitoring solutions. This how-to will introduce a simple setup for a single node configuration with Kata Containers, Prometheus and Grafana.

Reviewer: /open sources/open source/

> Prometheus and `Grafana` are open sources analytics & monitoring solution. This how-to will introduce a simple setup for one Linux node configuration, all Kata Containers, Prometheus and `Grafana` are running on the same host.

Reviewer 2: ```suggestion
`Prometheus` and `Grafana` are open sources analytics & monitoring solutions. This how-to will introduce a simple setup for a single node configuration with `Kata Containers`, `Prometheus` and `Grafana`.
```

> **Warning**: This how-to is only for evaluation purpose, you **SHOULD NOT** running it in production.

* [Install Prometheus](#install-prometheus)
* [Install `Grafana`](#install-grafana)
* [Setup `Grafana`](#setup-grafana)
  * [Create `datasource`](#create-datasource)
  * [Import dashboard](#import-dashboard)

> Both Prometheus and `Grafana` can installed using pre-compiled distributions.

Reviewer 2: What back ticks for Grafana but now Prometheus?

Also:
```suggestion
Both `Prometheus` and `Grafana` can be  installed using pre-compiled distributions.
```

Author: It's for the CI issues... ;-)

But now I have added Grafana to spelling check word list, so I removed all back ticks around Grafana.

Reviewer 2: Thanks

> ## Install Prometheus

Reviewer 2: I wonder if its worth doing this standalone example, versus enabling in Kubernetes. Do you expect folks to be running Prom/Grafana directly on the host, as opposed to daemonset in Kube? Is this just for a simple example?

Reviewer 2: Ah, I see you have the other document. I question if it is worth having both (some duplication of docs), but am okay with keeping this.  What do others think?

Reviewer: I would rather have both those in  single document.  I already have the feedback that we need to click on too many links to get to documentation, so I would rather have it as a single document. 
I would also have a link of the document in the Kata 2.0 metrics design doc.

Download and extract Prometheus.

```
$ export VERSION=2.18.1
$ curl -sL https://github.com/prometheus/prometheus/releases/download/v${VERSION}/prometheus-${VERSION}.linux-amd64.tar.gz -o prometheus-${VERSION}.linux-amd64.tar.gz
$ tar xvfz prometheus-${VERSION}.linux-amd64.tar.gz
$ cd prometheus-${VERSION}.linux-amd64
```

> To let Prometheus to collect metrics from Kata Containers, first add `magent` to a Prometheus target.

Reviewer 2: To enable Promethus collection of Kata Container metrics, add `magent` as a `Prometheus` target.

Open `prometheus.yml` and add `localhost:8090` to `scrape_configs` -> `static_configs`:

```yaml
scrape_configs:
  - job_name: 'prometheus'
    static_configs:
      - targets: ['localhost:8090']
```

And then start Prometheus server:

```
$ ./prometheus --config.file=prometheus.yml
```

> Prometheus listen on address `*:9000` by default.

Reviewer 2: ```suggestion
Prometheus listens on address `*:9000` by default.
```

## Install `Grafana`

In another terminal session, download and extract `Grafana`.

```
$ export VERSION=7.0.1
$ curl -sL https://dl.grafana.com/oss/release/grafana-${VERSION}.linux-amd64.tar.gz -o grafana-${VERSION}.linux-amd64.tar.gz
$ tar -zxvf grafana-${VERSION}.linux-amd64.tar.gz
$ cd grafana-${VERSION}
$ bin/grafana-server
```

## Setup `Grafana`

> After `Grafana` server started, you can open http://localhost:3000 to access `Grafana`. For `Grafana` 7.0.1, the default user/password is `admin/admin`. You can change or disable default admin account by modifying [configuration](https://grafana.com/docs/grafana/latest/installation/configuration/#security) , or even do more security settings.

Reviewer 2: ```suggestion
After the `Grafana` server is started, you can open http://localhost:3000 to access `Grafana`. For `Grafana` 7.0.1, the default user/password is `admin/admin`. You can modify the default account and adjust other security settings by editing the [grafana configuration](https://grafana.com/docs/grafana/latest/installation/configuration/#security).
```

To use `Grafana` show data from Prometheus, you must create a Prometheus `datasource` and dashboard.

### Create `datasource`

Open http://localhost:3000/datasources/new in your browser, select Prometheus from time series databases list.

Normally you only need to set `URL` to http://localhost:9090 to let it work, and leave the name as `Prometheus` as default.

### Import dashboard

> Here is a [sample dashboard](data/dashboard.json) provided, you can import and evaluate with it.

Reviewer 2: ```suggestion
 A [sample dashboard](data/dashboard.json) for Kata Containers metrics is provided which can be imported to Grafana for evaluation.
```
